test/BUILD

```diff
@@ -449,3 +449,13 @@ cc_test(
         "@boost//:hana",
     ],
 )
+
+cc_test(
+    name = "variant2_test",
```

Comment:
Added this trivial test to make sure header visibility on dependencies for the new library was right at least.